File: Old/HtmlPrototype/game/gameplay/turn.js
function turn(player){
    var $player = player;
    var $actionQueue = new Array();

    this.tick = function(context){
        var playerResourceChanges = new Array();

        $.each(context.systems(), function(key, value) {
            if(value.owner() == $player.playerNumber()){

                var planetResourceChanges = value.progressOneTurn();
                $.each(planetResourceChanges, function(key, value) {
                    playerResourceChanges.push(value);
                });
            }
        });

        $.each(playerResourceChanges, function(key, value) {
            $player.modifyResources(value);
        });
    };

    this.actionQueue = function(){
        return $actionQueue;
    };

    this.commitTurn = function(){
        $.each($actionQueue, function(key, value) {
            value.execute();
        });

    };
}

File: StarBastard/StarBastard.Core/Gameplay/Turn.cs
using System.Collections.Generic;

namespace StarBastard.Core.Gameplay
{
    public class Turn
    {
        private readonly Player _player;
        private List<object> _actionQueue;

        public Turn(Player player)
        {
            _player = player;
            _actionQueue = new List<object>();
        }

        public void tick(GameContext context)
        {
            var playerResourceChanges = new List<ResourceDelta>();

            foreach (var value in context.Systems)
            {
                if (value.Owner == _player.PlayerNumber)
                {
                    var resourceChanges = value.ProgressOneTurn();
                    foreach (var change in resourceChanges)
                    {
                        playerResourceChanges.Add(change);
                    }
                }
            }

            foreach (var change in playerResourceChanges)
            {
                _player.ModifyResources(change);
            }
        }



        public void commitTurn()
        {


        }
    }
}